src/accel/cuda/GPUSolver.cu

```diff
@@ -176,6 +180,60 @@ protected:
 
 };
 
+/**
+ * @brief Compute the stabilizing flux
+ * @param scalar_flux the scalar flux in each FSR and energy group
+ * @param stabilizing_flux the array of stabilizing fluxes
+ */
+__global__ void computeStabilizingFluxOnDevice(FP_PRECISION* scalar_flux,
+                                               FP_PRECISION* stabilizing_flux)
+{
+  if (stabilization_type == GLOBAL)
+  {
+    int tid = threadIdx.x + blockIdx.x * blockDim.x;
+    double multiplicative_factor = 1.0/stabilization_factor - 1.0;
+    while (tid < num_FSRs * NUM_GROUPS)
+    {
+      stabilizing_flux[tid] = multiplicative_factor * scalar_flux[tid];
+      tid += blockDim.x * gridDim.x;
+    }
+  }
+  else
+  {
+    // TODO, write other stabilization types.
+    // Apparently this is the only way to throw an error from a kernel at the
+    // moment... inline PTX assembly. craziness, man
+    asm("trap;");
```

Comment:
stabilization type could be checked during initialization, maybe we dont need this ?

Reply:
apparently there's an assertionError tha can be used in CUDA kernels. Would work for our case
[link](https://stackoverflow.com/questions/50755717/triggering-a-runtime-error-within-a-cuda-kernel)

Reply:
Yeah, it's definitely not needed, true. It's always checked beforehand, and I had primarily written this to ensure I'd not made a mistake.

Reply:
Let's delete it then

Reply:
Ah whoops. Totally meant to make that change but didn't